From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Simon Belgers <123083599+SBelgers@users.noreply.github.com>
Date: Tue, 17 Feb 2026 20:41:01 +0100
Subject: [PATCH 4/4] Remove duplicate QI_API on static inline methods in
 ssl.hpp

These methods are `static inline` defined in the header. The enclosing
class is already marked QI_API, so the extra QI_API on each method
causes MSVC warnings about inconsistent DLL linkage.
---
 qi/messaging/ssl/ssl.hpp | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/qi/messaging/ssl/ssl.hpp b/qi/messaging/ssl/ssl.hpp
index 93683d7..4d3dd8f 100644
--- a/qi/messaging/ssl/ssl.hpp
+++ b/qi/messaging/ssl/ssl.hpp
@@ -166,7 +166,6 @@ struct QI_API PemPasswordCallback
   ///   passphrase.
   ///
   /// @see `PEM_read_bio_X509`
-  QI_API
   static PemPasswordCallback fromPassphrase(const char* passphrase) noexcept;
 };
 
@@ -398,7 +397,6 @@ public:
   ///
   /// @see `BIO_new_file`
   /// @throws `Error` if the BIO could not be created.
-  QI_API
   static BIO fromFile(boost::filesystem::path path, const char* mode = "r");
 
   /// Returns the memory data for this BIO.
@@ -475,7 +473,6 @@ public:
   ///
   /// @throws `Error` if the file could not be read or if a private key header was found but the
   /// content could not be read.
-  QI_API
   static inline boost::optional<PKey> privateFromPemFile(boost::filesystem::path path,
                                                          const char* mode = "r",
                                                          PemPasswordCallback cb = {})
@@ -547,7 +544,6 @@ public:
   ///
   /// @throws `Error` if the file could not be read or if a X509 certificate header was found but
   /// the content could not be read.
-  QI_API
   static inline boost::optional<Certificate> fromPemFile(boost::filesystem::path path,
                                                          const char* mode = "r",
                                                          bool trusted = false,
@@ -564,7 +560,6 @@ public:
   ///
   /// An optional PEM password callback may be passed to decrypt the content of the PEM.
   /// @throws `Error` if a X509 certificate header was found but the content could not be read.
-  QI_API
   static inline std::vector<Certificate> chainFromPemFile(boost::filesystem::path path,
                                                           PemPasswordCallback cb = {})
   {
-- 
2.38.1.windows.1

